gradle-plugins/src/integTest/resources/ivy/modern/expectedChildIvyDescriptorContents.txt

```diff
@@ -0,0 +1,30 @@
+  <configurations>
+    <conf name="compile" visibility="public"/>
+    <conf name="dataTemplate" visibility="public" extends="mainGeneratedDataTemplateApiElements,mainGeneratedDataTemplateRuntimeElements"/>
```

Comment:
This only exposes `dataTemplate`, but not `dataModel`. I'm not sure if this is totally necessary, though. The Pegasus plugin resolves PDL schemas via a `FileCollection` which is basically just the target project's `dataModel` or `testDataModel` configuration. Most use cases add a dependency on external PDL schemas via `dataModel`, but consume the publication's `dataTemplate` configuration.... I think `dataModel` extends `dataTemplate`. Based on your knowledge, is there any risk here?

Reply:
tl;dr no risk, but IMO the legacy code exposing `dataModel` was incorrect.

Longer description:

Legacy Gradle was kind of simple, where consumers use the same configuration names to consume and publish dependencies - think of `runtime` and `compile`.

In modern Gradle, things get trickier. Consumers use the `implementation` configuration to specify a dependency, Gradle then internally uses the `compileClasspath` to consume it, and finally publishes a derived `compile` configuration.

My point of making this comparison is that `dataTemplate` and `dataModel` have a similar relationship. From my observations of existing behavior, yes `dataModel` is how consumers specify a set of data templates to consume. When the same project published its metadata, only publishing the `dataTemplate` configuration is necessary. In other words, `dataModel` is an internal implementation detail of a pegasus module, and concealing it from consumers by not enumerating `dataModel` in the Ivy descriptor is correct behavior. The legacy code exposing `dataModel` was incorrect.